hal/network/ncp_client/sara/sara_ncp_client.h

```diff
@@ -171,6 +171,10 @@ class SaraNcpClient: public CellularNcpClient {
     int waitAtResponseFromPowerOn(ModemState& modemState);
     int disablePsmEdrx();
     int checkSimReadiness(bool checkForRfReset = false);
+    int getOperationModeCached();
+    int setOperationModeCached(CellularOperationMode cemode);
+    int getOperationMode();
+    int setOperationMode(CellularOperationMode cemode, bool check = false, bool save = false);
```

Comment:
Currently It looks like we always save the operation mode to the cache. Can we simplify this function by omitting `save` parameter? 
Also, is there ever a situation where we dont want to `check` first? 
Can the logic in this function be simplified to 
1) Check operation mode
2) if it doesnt match, send the CEMODE command
3) Cache the new value
4) return

Reply:
We don't force the cache save in the registration intervening code, but need to force it on first boot.  I was going to optimize the check part as well, but decided against it because it makes the code less modular.